### PR TITLE
core: migrate_attachments: reduce number of threads

### PR DESCRIPTION
**Pull request by chaws**
Each thread might spawn multiple threads to make S3 uploads, thus causing OOM errors and `[WARNING] Connection pool is full, discarding connection` from urllib3.

Having 2 threads seems to be the sweet spot (tried with 4 and 8, but pod always gets killed)

### CODE REVIEW

File: squad/core/management/commands/migrate_attachments.py [hunk 1]
```diff
@@ -78,7 +77,7 @@ def handle(self, *args, **options):
             print('Nothing to do!')
             return
 
-        num_threads = multiprocessing.cpu_count() * 2
+        num_threads = 2
```
Comment (mwasilew) [inline squad/core/management/commands/migrate_attachments.py:80]: maybe this should be a parameter you can define in runtime. Default to 2.

Comment (chaws) [reply squad/core/management/commands/migrate_attachments.py]: Good idea! I've parameterized it

File: squad/core/management/commands/migrate_attachments.py [hunk 2]
```diff
@@ -65,20 +64,38 @@ def add_arguments(self, parser):
...
     def handle(self, *args, **options):
         show_progress = options['show_progress']
+        num_threads = options['num_threads']
+        groups = options['groups']
 
         null_storage_files = Q(tests_file_storage__isnull=True) | Q(metrics_file_storage__isnull=True) | Q(log_file_storage__isnull=True)
-        testrun_ids = TestRun.objects.filter(null_storage_files).order_by('id').values_list('id', flat=True)
+        queryset = TestRun.objects.filter(null_storage_files)
+
+        if groups != 'all':
```
Comment (mwasilew) [inline squad/core/management/commands/migrate_attachments.py:89]: 'all' is a valid group name. Maybe default should be empty list? You could use nargs=+ to specify that you're after a list.

**Review by mwasilew — CHANGES_REQUESTED** [review]

File: squad/core/management/commands/migrate_attachments.py (hunk 2 above)
Comment (chaws) [reply squad/core/management/commands/migrate_attachments.py:89]: Oh, I didn't think of it, let me fix it

Comment (chaws) [reply squad/core/management/commands/migrate_attachments.py]: Done, I've just make `default=''`

**Pull request merged**